dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/lifecycle/ProcessLifecycleMonitorTest.kt

```diff
@@ -0,0 +1,217 @@
+        underTest.onActivityStarted(mockActivity2)
+        underTest.onActivityResumed(mockActivity2)
+        underTest.onActivityStarted(mockActivity3)
+        underTest.onActivityResumed(mockActivity3)
+        underTest.onActivityPaused(mockActivity1)
+
+        // when
+        underTest.onActivityStopped(mockActivity1)
+        underTest.onActivityPaused(mockActivity2)
+        underTest.onActivityStopped(mockActivity2)
+        underTest.onActivityPaused(mockActivity3)
+        underTest.onActivityStopped(mockActivity3)
+
+        // then
+        val inOrder = inOrder(mockCallback)
+        inOrder.verify(mockCallback).onPaused()
+        inOrder.verify(mockCallback).onStopped()
+    }
+
+    @Test
+    fun `does not call onStopped onPaused if the activity was not resumed`() {
+        underTest.onActivityPaused(mockActivity1)
+        // when
+        underTest.onActivityStopped(mockActivity1)
+
+        // then
+        verifyZeroInteractions(mockCallback)
+    }
+
+    @Test
+    fun `does not call onStopped if onPause was not call upfront`() {
+        // given
+        underTest.onActivityStarted(mockActivity1)
+        underTest.onActivityResumed(mockActivity1)
+
+        // when
+        underTest.onActivityStopped(mockActivity1)
+
+        // then
+        verify(mockCallback, never()).onStopped()
+    }
+
+    @Test
+    fun `when starting activities from 2 different threads will only call onResumed once`() {
+
+        // given
+        val countDownLatch = CountDownLatch(2)
+
+        // when
+        Thread {
+            underTest.onActivityStarted(mockActivity1)
+            underTest.onActivityResumed(mockActivity1)
+            countDownLatch.countDown()
+        }.start()
+        Thread {
+            underTest.onActivityStarted(mockActivity2)
+            underTest.onActivityResumed(mockActivity2)
+            countDownLatch.countDown()
+        }.start()
+
+        countDownLatch.await()
+
+        // then
+        verify(mockCallback).onStarted()
+        verify(mockCallback).onResumed()
+    }
+
+    @Test
+    fun `when stopped from 2 different threads will only call onStooped once`() {
+
+        // given
+        underTest.onActivityStarted(mockActivity1)
+        underTest.onActivityResumed(mockActivity1)
+        underTest.onActivityStarted(mockActivity2)
+        underTest.onActivityResumed(mockActivity2)
+        val countDownLatch = CountDownLatch(2)
+
+        // when
+        Thread {
+            underTest.onActivityPaused(mockActivity1)
+            underTest.onActivityStopped(mockActivity1)
+            countDownLatch.countDown()
+        }.start()
+        Thread {
+            underTest.onActivityPaused(mockActivity2)
+            underTest.onActivityStopped(mockActivity2)
+            countDownLatch.countDown()
+        }.start()
+
+        countDownLatch.await()
+
+        // then
+        verify(mockCallback).onPaused()
+        verify(mockCallback).onStopped()
```

Comment:
Maybe you could add a verifyNoMoreInteractions(mockCallback) on all the test methods here to also make sure we don't have unwanted calls ?